From 817d304b3141d075efa5febb8f64f835a908a94f Mon Sep 17 00:00:00 2001
From: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
Date: Wed, 29 Jun 2016 13:39:37 +0700
Subject: [PATCH] add pwm pin function controller setting for r8a7745 chip

This patch file add pin function control of PWM Timer module for r8a7745 chip
on RZ/G1E Starter Kit

Signed-off-by: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
---
 drivers/pinctrl/sh-pfc/pfc-r8a7745.c | 167 +++++++++++++++++++++++++++++++++++
 1 file changed, 167 insertions(+)

diff --git a/drivers/pinctrl/sh-pfc/pfc-r8a7745.c b/drivers/pinctrl/sh-pfc/pfc-r8a7745.c
index 572827b..c7c1beb 100644
--- a/drivers/pinctrl/sh-pfc/pfc-r8a7745.c
+++ b/drivers/pinctrl/sh-pfc/pfc-r8a7745.c
@@ -1273,6 +1273,111 @@ static const unsigned int audio_clkout_c_pins[] = {
 static const unsigned int audio_clkout_c_mux[] = {
 	AUDIO_CLKOUT_C_MARK,
 };
+
+/* - PWM -------------------------------------------------------------------- */
+static const unsigned int pwm0_pins[] = {
+        RCAR_GP_PIN(1, 14),
+};
+static const unsigned int pwm0_mux[] = {
+        PWM0_MARK,
+};
+static const unsigned int pwm0_b_pins[] = {
+        RCAR_GP_PIN(5, 3),
+};
+static const unsigned int pwm0_b_mux[] = {
+        PWM0_B_MARK,
+};
+static const unsigned int pwm1_pins[] = {
+        RCAR_GP_PIN(4, 5),
+};
+static const unsigned int pwm1_mux[] = {
+        PWM1_MARK,
+};
+static const unsigned int pwm1_b_pins[] = {
+        RCAR_GP_PIN(5, 10),
+};
+static const unsigned int pwm1_b_mux[] = {
+        PWM1_B_MARK,
+};
+static const unsigned int pwm1_c_pins[] = {
+        RCAR_GP_PIN(1, 18),
+};
+static const unsigned int pwm1_c_mux[] = {
+        PWM1_C_MARK,
+};
+static const unsigned int pwm2_pins[] = {
+        RCAR_GP_PIN(4, 10),
+};
+static const unsigned int pwm2_mux[] = {
+        PWM2_MARK,
+};
+static const unsigned int pwm2_b_pins[] = {
+        RCAR_GP_PIN(5, 17),
+};
+static const unsigned int pwm2_b_mux[] = {
+        PWM2_B_MARK,
+};
+static const unsigned int pwm2_c_pins[] = {
+        RCAR_GP_PIN(0, 13),
+};
+static const unsigned int pwm2_c_mux[] = {
+        PWM2_C_MARK,
+};
+static const unsigned int pwm3_pins[] = {
+        RCAR_GP_PIN(4, 13),
+};
+static const unsigned int pwm3_mux[] = {
+        PWM3_MARK,
+};
+static const unsigned int pwm3_b_pins[] = {
+        RCAR_GP_PIN(0, 16),
+};
+static const unsigned int pwm3_b_mux[] = {
+        PWM3_B_MARK,
+};
+static const unsigned int pwm4_pins[] = {
+        RCAR_GP_PIN(1, 3),
+};
+static const unsigned int pwm4_mux[] = {
+        PWM4_MARK,
+};
+static const unsigned int pwm4_b_pins[] = {
+        RCAR_GP_PIN(0, 21),
+};
+static const unsigned int pwm4_b_mux[] = {
+        PWM4_B_MARK,
+};
+static const unsigned int pwm5_pins[] = {
+        RCAR_GP_PIN(3, 30),
+};
+static const unsigned int pwm5_mux[] = {
+        PWM5_MARK,
+};
+static const unsigned int pwm5_b_pins[] = {
+        RCAR_GP_PIN(4, 0),
+};
+static const unsigned int pwm5_b_mux[] = {
+        PWM5_B_MARK,
+};
+static const unsigned int pwm5_c_pins[] = {
+        RCAR_GP_PIN(0, 10),
+};
+static const unsigned int pwm5_c_mux[] = {
+        PWM5_C_MARK,
+};
+static const unsigned int pwm6_pins[] = {
+        RCAR_GP_PIN(4, 8),
+};
+static const unsigned int pwm6_mux[] = {
+        PWM6_MARK,
+};
+static const unsigned int pwm6_b_pins[] = {
+        RCAR_GP_PIN(0, 7),
+};
+static const unsigned int pwm6_b_mux[] = {
+        PWM6_B_MARK,
+};
+
 /* - DU --------------------------------------------------------------------- */
 static const unsigned int du0_rgb666_pins[] = {
 	/* R[7:2], G[7:2], B[7:2] */
@@ -2866,6 +2971,23 @@ static const struct sh_pfc_pin_group pinmux_groups[] = {
 	SH_PFC_PIN_GROUP(audio_clkout),
 	SH_PFC_PIN_GROUP(audio_clkout_b),
 	SH_PFC_PIN_GROUP(audio_clkout_c),
+        SH_PFC_PIN_GROUP(pwm0),
+        SH_PFC_PIN_GROUP(pwm0_b),
+        SH_PFC_PIN_GROUP(pwm1),
+        SH_PFC_PIN_GROUP(pwm1_b),
+        SH_PFC_PIN_GROUP(pwm1_c),
+        SH_PFC_PIN_GROUP(pwm2),
+        SH_PFC_PIN_GROUP(pwm2_b),
+        SH_PFC_PIN_GROUP(pwm2_c),
+        SH_PFC_PIN_GROUP(pwm3),
+        SH_PFC_PIN_GROUP(pwm3_b),
+        SH_PFC_PIN_GROUP(pwm4),
+        SH_PFC_PIN_GROUP(pwm4_b),
+        SH_PFC_PIN_GROUP(pwm5),
+        SH_PFC_PIN_GROUP(pwm5_b),
+        SH_PFC_PIN_GROUP(pwm5_c),
+        SH_PFC_PIN_GROUP(pwm6),
+        SH_PFC_PIN_GROUP(pwm6_b),
 	SH_PFC_PIN_GROUP(du0_rgb666),
 	SH_PFC_PIN_GROUP(du0_rgb888),
 	SH_PFC_PIN_GROUP(du1_rgb666),
@@ -3101,6 +3223,44 @@ static const char * const du_groups[] = {
 	"du1_cde_disp",
 };
 
+static const char * const pwm0_groups[] = {
+        "pwm0",
+        "pwm0_b",
+};
+
+static const char * const pwm1_groups[] = {
+        "pwm1",
+        "pwm1_b",
+        "pwm1_c",
+};
+
+static const char * const pwm2_groups[] = {
+        "pwm2",
+	"pwm2_b",
+	"pwm2_c",
+};
+
+static const char * const pwm3_groups[] = {
+        "pwm3",
+	"pwm3_b",
+};
+
+static const char * const pwm4_groups[] = {
+        "pwm4",
+        "pwm4_b",
+};
+
+static const char * const pwm5_groups[] = {
+        "pwm5",
+        "pwm5_b",
+	"pwm5_c",
+};
+
+static const char * const pwm6_groups[] = {
+        "pwm6",
+	"pwm6_b",
+};
+
 static const char * const du0_groups[] = {
 	"du0_clk_in",
 };
@@ -3393,6 +3553,13 @@ static const char * const vin1_groups[] = {
 
 static const struct sh_pfc_function pinmux_functions[] = {
 	SH_PFC_FUNCTION(audio_clk),
+        SH_PFC_FUNCTION(pwm0),
+        SH_PFC_FUNCTION(pwm1),
+        SH_PFC_FUNCTION(pwm2),
+        SH_PFC_FUNCTION(pwm3),
+        SH_PFC_FUNCTION(pwm4),
+        SH_PFC_FUNCTION(pwm5),
+        SH_PFC_FUNCTION(pwm6),
 	SH_PFC_FUNCTION(du),
 	SH_PFC_FUNCTION(du0),
 	SH_PFC_FUNCTION(du1),
-- 
1.9.1

